linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -310,31 +312,47 @@ object Infos {
       this
     }
 
-    def addAccessedClassData(tpe: TypeRef): this.type =
-      addAccessedClassData(baseNameOf(tpe))
+    def maybeAddAccessedClassData(typeRef: TypeRef): this.type = {
+      typeRef match {
+        case ClassRef(cls) =>
+          addAccessedClassData(cls)
+        case ArrayTypeRef(ClassRef(cls), _) =>
+          addAccessedClassData(cls)
+        case _ =>
+      }
+      this
```

Comment:
Create a helper for this pattern match? It appears many times.

Something like:

```scala
private def maybeAdd(typeRef: TypeRef, add: String => this.type): this.type = {
 // ...
}
```